### PR TITLE
feat(gitlab): Job retry on `agent.version` and `pipeline.changelog`

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Added an `Exit(code=42)` on python invoke tasks calling `aws` cli during the pipeline.
And throw that exit code in the gitlab script call (using `command || exit $?`) as this is currently mandatory for gitlab to get the correct error code.

### Motivation
[acix-435](https://datadoghq.atlassian.net/browse/ACIX-435) - Follow-up of #29262. In some situation the first call to `aws` cli is not a call to `aws ssm` but `aws cp`. With addition of the `Exit(code=42)`, if these tasks fail the job will be retried

### Describe how to test/QA your changes
We can only validate I didn't change the current behaviour. We can validate with a log search of looking at retried job executions, which is tedious. Otherwise we can just monitor if any "Unable to locate credential" issue re-occurs in the pipeline.

### Possible Drawbacks / Trade-offs
In some modified calls there are still some other `aws` commands called first, this will come in a next PR

### Additional Notes
I took the opportunity to remove unnecessary `export` keywords

### CODE REVIEW

File: .gitlab/deploy_packages/nix.yml [hunk 1]
```diff
@@ -185,7 +185,7 @@ deploy_staging_iot_agent:
   needs: ["build_iot_agent-binary_x64"]
   script:
     - $S3_CP_CMD $S3_ARTIFACTS_URI/iot/agent ./agent
```
Comment (chouetz) [inline .gitlab/deploy_packages/nix.yml:187]: Here the script can fail on that line. I did the modification below for consistency and will do a follow-up PR for such calls

**Review by amenasria — APPROVED** [review]
LGTM!

**Review by clarkb7 — APPROVED** [review]

File: .gitlab/packaging/oci.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
   tags: ["arch:amd64"]
   before_script:
-    - export PACKAGE_VERSION=$(inv agent.version --url-safe --major-version 7)-1
+    - PACKAGE_VERSION="$(inv agent.version --url-safe --major-version 7)-1" || exit $?
```
Comment (AliDatadog) [inline .gitlab/packaging/oci.yml:9]: Can we use `PACKAGE_VERSION` in `script` if it's not exported ?

**Review by NouemanKHAL — APPROVED** [review]

File: .gitlab/packaging/oci.yml (hunk 2 above)
Comment (AliDatadog) [reply .gitlab/packaging/oci.yml:9]: Interesting ! Thanks for sharing

**Review by AliDatadog — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]

**Pull request merged**